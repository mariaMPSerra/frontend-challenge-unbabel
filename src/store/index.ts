import { getTranscriptions } from '@/api/getTranscriptions'
import { updateTranscriptions } from '@/api/updateTranscriptions'
import { defineStore } from 'pinia'

const transcriptionsInitialState = {
  data: []
}

export const useTranscriptionsStore = defineStore('transcriptions', {
  state: () => transcriptionsInitialState,
  actions: {
    async getTranscriptionsData() {
      try {
        const response = await getTranscriptions()

        this.data = response
      } catch (error) {
        return console.error('Error fetching home data:', error)
      }
    },
    async updateTranscriptionsData() {
      try {
        const response = await updateTranscriptions(this.data)

        return response
      } catch (error) {
        console.error('Error update data:', error)
      }
    },
    updateTranscriptionsItems(itemData, type) {
      if (type === 'add') {
        this.data?.push(itemData)
      } else {
        this.data = this.data?.filter((item) => item.id !== itemData)
      }
    },
    addNewTranscriptionsItem(newItem) {
      this.data.push(newItem)
    }
  }
})
