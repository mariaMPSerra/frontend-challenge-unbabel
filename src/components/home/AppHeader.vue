<template>
  <div class="transcriptions__header">
    <div class="transcriptions__header__content">
      <h3 class="transcriptions__header__title">Transcriptions</h3>
      <div class="transcriptions__header__actions">
        <button @click="updateTranscriptions" :disabled="!data?.length">
          <img src="@/assets/icons/upload.svg" class="transcriptions__header__btn" />
        </button>
        <button @click="getTranscriptionsData">
          <img src="@/assets/icons/fetch-document.svg" class="transcriptions__header__btn" />
        </button>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { useTranscriptionsStore } from '@/store'
import { storeToRefs } from 'pinia'

const transcriptionsStore = useTranscriptionsStore()
const { getTranscriptionsData, updateTranscriptionsData } = transcriptionsStore
const { data } = storeToRefs(transcriptionsStore)

const updateTranscriptions = async () => {
  await updateTranscriptionsData()
}
</script>

<style lang="scss">
.transcriptions__header {
  display: flex;
  background: $white;
  box-shadow: 0px 1px 2px $gray;
  margin-bottom: 20px;
  padding: 15px 0;
  width: 100%;

  &__content {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-left: auto;
    margin-right: auto;
    min-width: 75%;
  }

  &__btn {
    cursor: pointer;

    &:first-child {
      @media (min-width: 600px) {
        margin-right: 25px;
      }
    }
  }

  &__title {
    font-family: 'Montserrat', sans-serif;
    font-size: 18px;
    font-weight: 500;
    color: $dark-purple;
    margin: 0 0 0 15px;
  }

  &__actions {
    margin-right: 15px;
  }
}
</style>
