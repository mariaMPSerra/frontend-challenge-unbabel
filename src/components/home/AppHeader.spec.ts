import { beforeEach, describe, expect, it, vi } from 'vitest'
import AppHeader from './AppHeader.vue'
import { flushPromises, mount } from '@vue/test-utils'
import { createPinia, setActivePinia } from 'pinia'
import { useTranscriptionsStore } from '@/store'

describe('AppHeader Component', () => {
  beforeEach(() => {
    setActivePinia(createPinia())
  })

  it('renders AppHeader component', async () => {
    const wrapper = mount(AppHeader)

    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('methods', () => {
    it('updateTranscriptions', async () => {
      const transcriptionsStore = useTranscriptionsStore()

      vi.spyOn(transcriptionsStore, 'updateTranscriptionsData')
      vi.spyOn(window, 'alert').mockImplementation(() => {})

      const wrapper = mount(AppHeader)

      await wrapper.vm.updateTranscriptions()

      await flushPromises()

      expect(transcriptionsStore.updateTranscriptionsData).toHaveBeenCalled()
      expect(window.alert).toHaveBeenCalled()
    })
  })
})
