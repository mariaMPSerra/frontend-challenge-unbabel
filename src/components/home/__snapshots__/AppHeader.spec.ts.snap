// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHome Component > renders AppHome component 1`] = `
"<div class="transcriptions__header">
  <div class="transcriptions__header__content">
    <h3 class="transcriptions__header__title">Transcriptions</h3>
    <div class="transcriptions__header__actions"><button disabled=""><img src="/src/assets/icons/upload.svg" class="transcriptions__header__btn"></button><button><img src="/src/assets/icons/fetch-document.svg" class="transcriptions__header__btn"></button></div>
  </div>
</div>"
`;
